tools/slang-test/main.cpp

```diff
@@ -1830,25 +1830,72 @@ TestResult skipTest(TestContext* /* context */, TestInput& /*input*/)
     return kTestResult_Ignored;
 }
 
-static bool hasD3D12Option(const TestOptions& testOptions)
+
+static bool hasRenderOption(RenderApiType apiType, const List<String>& options)
```

Comment:
This is a little ugly, to be manually testing for tests that are already specifying a renderer. Eventually we should figure out the Right Way to have tests get registered for per-API execution (with the possibility that they will only be enabled for a subset of APIs) and then switch everything to use that model, whatever it is...

Reply:
Agreed, this should be done in a more robust manner.